lib/internal/crypto/pbkdf2.js

```diff
@@ -34,10 +38,40 @@ function _pbkdf2(password, salt, iterations, keylen, digest, callback) {
   password = toBuf(password);
   salt = toBuf(salt);
 
+  if (!isArrayBufferView(password)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'password',
+                               ['string', 'Buffer', 'TypedArray']);
+  }
+
+  if (!isArrayBufferView(salt)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'salt',
+                               ['string', 'Buffer', 'TypedArray']);
+  }
+
+  if (typeof iterations !== 'number')
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'iterations', 'number');
+
+  if (iterations < 0)
+    throw new errors.RangeError('ERR_OUT_OF_RANGE', 'iterations');
+
+  if (typeof keylen !== 'number')
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'keylen', 'number');
+
+  if (keylen < 0 ||
+      !Number.isFinite(keylen) ||
+      isNaN(keylen) ||
```

Comment:
`Number.isFinite` already checks if it is a `NaN`

Reply:
ah right, good point :-)